filament/src/details/RenderTarget.cpp

```diff
@@ -105,28 +116,42 @@ RenderTarget* RenderTarget::Builder::build(Engine& engine) {
     uint32_t maxWidth = 0;
     uint32_t minHeight = std::numeric_limits<uint32_t>::max();
     uint32_t maxHeight = 0;
-    uint32_t minDepth = std::numeric_limits<uint32_t>::max();
-    uint32_t maxDepth = 0;
+    uint32_t layerCountMultiview = 0;
     for (auto const& attachment : mImpl->mAttachments) {
         if (attachment.texture) {
             const uint32_t w = attachment.texture->getWidth(attachment.mipLevel);
             const uint32_t h = attachment.texture->getHeight(attachment.mipLevel);
             const uint32_t d = attachment.texture->getDepth(attachment.mipLevel);
+            if (attachment.layerCount) {
+                FILAMENT_CHECK_PRECONDITION(
+                        attachment.texture->getTarget() == Texture::Sampler::SAMPLER_2D_ARRAY)
+                        << "Texture sampler must be of 2d array for multiview";
+                FILAMENT_CHECK_PRECONDITION(attachment.layer + attachment.layerCount <= d)
+                        << "For multiview, layer + layerCount cannot exceed the number of depth";
+                if (layerCountMultiview) {
+                    FILAMENT_CHECK_PRECONDITION(layerCountMultiview == attachment.layerCount)
+                            << "layerCount for multiview should match";
+                } else {
+                    layerCountMultiview = attachment.layerCount;
+                }
+            }
```

Comment:
instead of doing this, I would calculate the min and max layerCount and do the checks outside of the loop:

CHECK( min == max ) // all counts must match
CHECK( min > 1 && SAMPLER_2D || min <= 1 ) // multiview only with sampler 2D

The (attachment.layer + attachment.layerCount <= d) check can be always done (even when attachment.layerCount ==0).